scripts/src/public/services/orders.ts

```diff
@@ -250,14 +277,15 @@ export async function submitOrder(
 
 		if (order.type === "earn") {
 			const offerContent = (await offerContents.getOfferContent(order.offerId, logger))!;
+
 			switch (offerContent.contentType) {
 				// TODO this switch-case should be inside the offerContents module
 				case "poll":
 					// validate form
 					if (!offerContents.isValid(offerContent, form)) {
 						throw InvalidPollAnswers();
 					}
-					await offerContents.savePollAnswers(order.recipient!.id, order.offerId, orderId, form); // TODO should we also save quiz results?
+					await offerContents.savePollAnswers(order.user.id, order.offerId, orderId, form); // TODO should we also save quiz results?
```

Comment:
why not recipient?

Reply:
the `order.isMarketplaceOrder()` type guard, when returning true, indicates that order is of type `MarketplaceOrder`.
that type always has only one user, so it exposes that `user` property.
there's currently no type guard for `isEarn()` which will provide a `recipient` property (it does exists for `isP2P()`)